comixed-services/src/main/java/org/comixedproject/service/scraping/ScrapingService.java

```diff
@@ -0,0 +1,254 @@
+ * aLong with this program. If not, see <http://www.gnu.org/licenses>
+ */
+
+package org.comixedproject.service.scraping;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.extern.log4j.Log4j2;
+import org.comixedproject.model.comic.Comic;
+import org.comixedproject.model.comic.Credit;
+import org.comixedproject.scrapers.ScrapingException;
+import org.comixedproject.scrapers.comicvine.adaptors.ComicVineScrapingAdaptor;
+import org.comixedproject.scrapers.model.ScrapingIssue;
+import org.comixedproject.scrapers.model.ScrapingIssueDetails;
+import org.comixedproject.scrapers.model.ScrapingVolume;
+import org.comixedproject.service.comic.ComicException;
+import org.comixedproject.service.comic.ComicService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * <code>ScrapingService</code> handles interacting with the selected {@link ScrapingAdaptor} and
+ * the scraping cache.
+ *
+ * @author Darryl L. Pierce
+ */
+@Service
+@Log4j2
+public class ScrapingService {
+  @Autowired private ComicVineScrapingAdaptor scrapingAdaptor;
+  @Autowired private ObjectMapper objectMapper;
+  @Autowired private ScrapingCacheService scrapingCacheService;
+  @Autowired private ComicService comicService;
+
+  /**
+   * Retrieves a list of volumes for the given series, up to the max records specified.
+   *
+   * @param apiKey the api key
+   * @param series the series name
+   * @param maxRecords the maximum records
+   * @param skipCache the skip cache flag
+   * @return the volumes
+   * @throws ScrapingException if an error occurs
+   */
+  public List<ScrapingVolume> getVolumes(
+      final String apiKey, final String series, final Integer maxRecords, final boolean skipCache)
+      throws ScrapingException {
+    List<ScrapingVolume> result = new ArrayList<>();
+    final String source = this.scrapingAdaptor.getSource();
+    final String key = this.scrapingAdaptor.getVolumeKey(series);
+
+    log.debug(
+        "Getting volumes: series={} maxRecords={} skipCache={}", series, maxRecords, skipCache);
+    if (!skipCache) {
+      log.debug("Fetching from the cache: source={} key={}", source, key);
+      final List<String> cachedEntries = this.scrapingCacheService.getFromCache(source, key);
+      if (cachedEntries != null && !cachedEntries.isEmpty()) {
+        log.debug(
+            "Decoding {} entr{}", cachedEntries.size(), cachedEntries.size() == 1 ? "y" : "ies");
+        for (int index = 0; index < cachedEntries.size(); index++) {
+          try {
+            result.add(this.objectMapper.readValue(cachedEntries.get(index), ScrapingVolume.class));
+          } catch (JsonProcessingException error) {
+            throw new ScrapingException("Failed to decode scraping volume", error);
+          }
+        }
```

Comment:
Wouldn't it be possible to use an enhanced for-loop ?

Reply:
Yeah, it's an old habit to work with indexes. 😄